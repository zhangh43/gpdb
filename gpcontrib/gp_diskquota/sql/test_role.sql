-- Test role quota

CREATE SCHEMA srole;
SET search_path TO srole;

CREATE ROLE u1 NOLOGIN;
CREATE ROLE u2 NOLOGIN;
CREATE TABLE b (t TEXT);
ALTER TABLE b OWNER TO u1;
CREATE TABLE b2 (t TEXT);
ALTER TABLE b2 OWNER TO u1;

SELECT diskquota.set_role_quota('u1', '1 MB');

INSERT INTO b SELECT generate_series(1,100);
-- expect insert fail
INSERT INTO b SELECT generate_series(1,100000000);
-- expect insert fail
INSERT INTO b SELECT generate_series(1,100);
-- expect insert fail
INSERT INTO b2 SELECT generate_series(1,100);
ALTER TABLE b OWNER TO u2;
SELECT pg_sleep(5);
-- expect insert succeed
INSERT INTO b SELECT generate_series(1,100);
-- expect insert succeed
INSERT INTO b2 SELECT generate_series(1,100);

DROP TABLE b, b2;
DROP ROLE u1, u2;
RESET search_path;
DROP SCHEMA srole;
