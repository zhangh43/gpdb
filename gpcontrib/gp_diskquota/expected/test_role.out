-- Test role quota
CREATE SCHEMA srole;
SET search_path TO srole;
CREATE ROLE u1 NOLOGIN;
NOTICE:  resource queue required -- using default resource queue "pg_default"
CREATE ROLE u2 NOLOGIN;
NOTICE:  resource queue required -- using default resource queue "pg_default"
CREATE TABLE b (t TEXT);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 't' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
ALTER TABLE b OWNER TO u1;
CREATE TABLE b2 (t TEXT);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 't' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
ALTER TABLE b2 OWNER TO u1;
SELECT diskquota.set_role_quota('u1', '1 MB');
 set_role_quota 
----------------
 
(1 row)

INSERT INTO b SELECT generate_series(1,100);
-- expect insert fail
INSERT INTO b SELECT generate_series(1,100000000);
ERROR:  role's disk space quota exceeded with name:u1
-- expect insert fail
INSERT INTO b SELECT generate_series(1,100);
ERROR:  role's disk space quota exceeded with name:u1
-- expect insert fail
INSERT INTO b2 SELECT generate_series(1,100);
ERROR:  role's disk space quota exceeded with name:u1
ALTER TABLE b OWNER TO u2;
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- expect insert succeed
INSERT INTO b SELECT generate_series(1,100);
-- expect insert succeed
INSERT INTO b2 SELECT generate_series(1,100);
DROP TABLE b, b2;
DROP ROLE u1, u2;
RESET search_path;
DROP SCHEMA srole;
